internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -448,21 +456,31 @@ func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	if err == nil {
 		return nil
 	}
+	if errors.Is(err, context.DeadlineExceeded) ||
+		errors.Is(err, context.Canceled) {
+		return fmt.Errorf("could not reload daemon: %w", err)
+	}
 
 	signal := make(chan struct{})
+	defer close(signal)
 	backExp := backoff.NewExpBackoff(signal, 10*time.Second, 1*time.Minute)
 
-	for i := 5; i >= 0; i-- {
+	for i := 0; i < 5; i++ {
 		backExp.Wait()
 		c.log.Info("Retrying to restart...")
+
```

Comment:
I can't comment on the line `c.log.Info("Retrying to restart...")` above since it didn't change, but we should log which retry this is and what the last error was.

Reply:
We aren't logging the error message from failed retries right now.